File: Python/leetcode_594.py
from collections import Counter
from typing import List


def find_lhs(nums: List[int]) -> int:
    c = Counter(nums)
    max_l = 0
    for n in c:
        if n+1 in c:
            max_l = max(max_l, c[n] + c[n+1])
    return max_l


print(find_lhs([1, 3, 5, 4, 5, 2, 3, 7]))

File: JavaScript/leetcode_594.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findLHS = function (nums) {
    let maxl = 0;
    const freqMap = new Map();
    for (const item of nums) {
        freqMap.set(item, (freqMap.get(item) || 0) + 1);
    }
    for (const key of freqMap.keys()) {
        if (freqMap.has(key + 1)) {
            maxl = Math.max(maxl, freqMap.get(key) + freqMap.get(key + 1));
        }
    }
    return maxl;
};
findLHS([1, 2, 2]);
